src/gromacs/nblib/interactions.h

```diff
@@ -46,18 +46,67 @@
 
 #include "gromacs/math/vectypes.h"
 
+#include <string>
+#include <vector>
+
 namespace nblib
 {
 
+using BondName = std::string;
+using EquiDist = real;
+using ForceConstant = real;
+
 //! Type of interaction
-struct HarmonicType
+struct HarmonicBondType
 {
     real equiDist;
     real forceConstant;
-    std::string atomIDi;
-    std::string atomIDj;
-    //int atomI;
-    //int atomJ;
+    BondName bondName;
+
+    HarmonicBondType(EquiDist ed, ForceConstant fc, BondName name)
+    : equiDist(ed), forceConstant(fc), bondName(std::move(name)) {}
+
+    //! Force explicit use of correct types
+    template<typename T, typename U, typename V>
+    HarmonicBondType(T ed, U fc, V name) = delete;
+};
+
+//! Type of interaction for storage
+struct Interaction
+{
+    void					 *interactionType;
+    size_t					  atomIndex1;
+    size_t					  atomIndex2;
+
+    Interaction (void* interactionPointer, size_t atomIndex1, size_t atomIndex2)
```

Comment:
We should avoid the use of void pointers, as they are impossible to debug.

Reply:
Correct. 

I wanted to have a handle to the bond type with its params, potential functions, etc without explicitly storing an instance of it within the Interaction struct.

Another way I could have accomplished it was by using Inheritance. However, a weakly enforcing one that doesn't require virtual function or shared members.